### PR TITLE
feat(trading): cross chain deposits

### LINKED ISSUE #6109: Squid router UI build in portfolio page
## Story

As a user 
I want a seemless deposit experience from Squid router
So that I can deposit easily

## Acceptance Criteria

- [x] #6267 must be done to enable this
  - [ ] "Cross Chain Deposit"
    - [ ] Embed the full Squid router widget as per the spike
    - [ ] Customise the colour scheme to match Vega colours (Joelle can help here if needed, or just play around)
    - [ ] Should be "one click" from the squid router all the way to Vega

### Cross Chain

Will embed this full widget in that panel

image

### PR DESCRIPTION
**Pull request by asiaznik**
# Related issues 🔗

Closes #6109 

# Description ℹ️

* adds 'Cross chain deposits' tab to the portfolio page
* adds SquidRouter widget that handles the operation
* adds user configurable feature flags that can turn on/off the feature
* adds hardcoded arbitrum bridge and assets for testing purposes
* adds arbitrum bridge (SquidReceiver) ABI configs
* adds arbitrum bridge (SquidReceiver) contract

Run `NX_CROSS_CHAIN_DEPOSITS_ENABLED=true yarn env-cmd -f ./apps/trading/.env.mainnet nx serve trading` to test it locally.

# Demo 📺

Screenshot 2024-05-07 at 20 05 10
Screenshot 2024-05-07 at 20 05 17
Screenshot 2024-05-07 at 20 05 39
Screenshot 2024-05-08 at 12 59 54
Screenshot 2024-05-08 at 16 47 59

# Technical 👨‍🔧

Includes some hardcoded values (behind flag) that would need to be deleted once everything else is ready (actual bridge, etc).

Also doesn't ask for `chainId` property from the assets API as it doesn't exist yet on main net, re-gen needed once mainnet is up to date.

### CODE REVIEW

File: libs/smart-contracts/src/index.ts [hunk 1]
```diff
@@ -1,2 +1,5 @@
 export * from './contracts';
 export * from './utils';
+
+export { default as ERC20_ABI } from './abis/erc20_abi.json';
```
Comment (mattrussell36) [inline libs/smart-contracts/src/index.ts:4]: I think we can un export these now that we we use the class wrappers for each contract

File: libs/web3/src/lib/use-evm-bridge-configs.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+   */
+  name: string;
+};
+
+type EVMBridgeConfigs = {
+  configs: EVMBridgeConfig[];
+};
+
+export const useEVMBridgeConfigs = () => {
+  const { data, loading, error } = useNetworkParamsQuery();
```
Comment (mattrussell36) [inline libs/web3/src/lib/use-evm-bridge-configs.ts:62]: You might want to use the useNetworkParam hook rather than the raw query

File: apps/trading/pages/client-router.tsx [hunk 3]
```diff
@@ -196,7 +196,7 @@ export const useRouterConfig = (): RouteObject[] => {
     },
     {
       path: 'portfolio/*',
-      element: <Outlet />,
+      element: <LayoutWithSidebar sidebar={null} />,
```
Comment (mattrussell36) [inline apps/trading/pages/client-router.tsx:199]: Why this change? I ended up in this state (sidebar open without the option to toggle it off):

Screenshot 2024-05-14 at 14 58 51

**Comment by mattrussell36** [comment]
@asiaznik @daro-maj we'll need to adjust/hide this title

Screenshot 2024-05-14 at 15 14 55

**Comment by mattrussell36** [comment]
@asiaznik @daro-maj 

Until this is using the correct production bridge for mainnet we need to 'fully' feature flag this so there is no way for the users to enable it

**Review by daro-maj — APPROVED** [review]

**Review by MadalinaRaicu — APPROVED** [review]

**Review by jeremyletang — APPROVED** [review]

File: apps/trading/lib/hooks/use-squid-router-config.ts [hunk 4]
```diff
@@ -0,0 +1,466 @@
...
+const isSquidFriendlyAsset = (
+  asset: AssetFieldsFragment
+): asset is SquidFriendlyAsset =>
+  Boolean(
+    asset.source.__typename === 'ERC20' &&
+      asset.source.chainId &&
+      asset.source.contractAddress
+  );
+
+const HARDCODED_EVM_CONFIGS: EVMBridgeConfig[] = [
```
Comment (mattrussell36) [inline apps/trading/lib/hooks/use-squid-router-config.ts:72]: We should be able to un-hardcode these now

**Review by mattrussell36 — APPROVED** [review]
